mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -147,4 +157,46 @@ private boolean belongsToSameUser(Mailbox mailbox, Mailbox otherMailbox) {
     public void updateACL(Mailbox mailbox, MailboxACL.MailboxACLCommand mailboxACLCommand) throws MailboxException{
         mailbox.setACL(mailbox.getACL().apply(mailboxACLCommand));
     }
+
+    @Override
+    public List<MailboxAnnotation> getAnnotations(Mailbox mailbox) throws MailboxException {
+        InMemoryId maiboxId = (InMemoryId) mailbox.getMailboxId();
+        return Lists.newArrayList(retrieveAllAnnotations(maiboxId));
+    }
+
+    private synchronized Iterable<MailboxAnnotation> retrieveAllAnnotations(InMemoryId maiboxId) {
```

Comment:
I really don't like synchronized on `this`, anybody can deadlock the application with just lock your object from the outside.